overlays/nodejs/serverless/function.ts

```diff
@@ -309,17 +311,17 @@ function addPackageAndDependenciesToSet(s: Set<string>, root: Package, pkg: stri
 // It is assumed that the tree was correctly construted such that dependencies are resolved to compatible versions in
 // the closest available match starting at the provided root and walking up to the head of the tree.
 function findDependency(root: Package, name: string) {
-    for(;root;root = root.parent) {
+    for (; root; root = root.parent) {
         for (var child of root.children) {
             let childName = child.name;
             // Note: `read-package-tree` returns incorrect `.name` properties for packages in an orgnaization - like
             // `@types/express` or `@protobufjs/path`.  Compute the correct name from the `path` property instead.
             // Match any name that ends with something that looks like `@foo/bar`.
-            const match = /\/\@[^\/]*\/[^\/]*$/.exec(child.path);
+            const match = /\@[^\/]*\/[^\/]*$/.exec(child.path);
```

Comment:
I personally vote against using regexes to match paths like this.  I'd prefer to see us using real filesystem operations that parse the path more precisely.  (For example, this regex will now match `some@foo/bar`, which I'm not sure is intentional?  And if it is, I'd rather state this more precisely than just "happening" to match it using a regex.)

Reply:
Yes - we’ll move this over to real path operations next. Just opened this to verify that things were working correctly with the minimal change.